Tests/Common/Communication/EventStorageTests.swift

```diff
@@ -0,0 +1,165 @@
+    }
+
+    func test_loadEvents_givenCorruptDataInFile_expectValidEventsLoaded() async throws {
+        // Ensure the directory for the event type exists
+        let eventTypeDirectory = await eventStorageManager.baseDirectory.appendingPathComponent(ProfileIdentifiedEvent.key)
+        try FileManager.default.createDirectory(at: eventTypeDirectory, withIntermediateDirectories: true)
+
+        // Store a corrupt event file manually
+        let corruptData = Data("corrupt data".utf8)
+        let corruptFileURL = eventTypeDirectory.appendingPathComponent("corruptEvent.json")
+        try corruptData.write(to: corruptFileURL)
+
+        // Store a valid event file
+        let validEvent = ProfileIdentifiedEvent(identifier: "testID")
+        try await eventStorageManager.store(event: validEvent)
+
+        // Attempt to load events
+        let loadedEvents = try await eventStorageManager.loadEvents(ofType: ProfileIdentifiedEvent.key)
+        let containsValidEvent = loadedEvents.contains { ($0 as? ProfileIdentifiedEvent)?.identifier == "testID" }
+        XCTAssertTrue(containsValidEvent, "Valid events should be loaded even if corrupt data is present")
+    }
+
+    // MARK: - Event Removal Tests
+
+    func test_removeEvent_givenStoredEvent_expectEventRemoved() async throws {
+        let event = ProfileIdentifiedEvent(identifier: "testID")
+        try await eventStorageManager.store(event: event)
+
+        let eventFileURL = await eventStorageManager.baseDirectory
+            .appendingPathComponent(ProfileIdentifiedEvent.key)
+            .appendingPathComponent("\(event.storageId).json")
+        await eventStorageManager.remove(ofType: ProfileIdentifiedEvent.key, withStorageId: event.storageId)
+
+        let fileExists = FileManager.default.fileExists(atPath: eventFileURL.path)
+        XCTAssertFalse(fileExists, "Event file should not exist after removal")
+    }
+
+    func test_removeEvent_givenNonExistentFile_expectNoErrorThrown() async throws {
+        let eventType = ProfileIdentifiedEvent.key
+        let nonExistentStorageId = UUID().uuidString
+        await eventStorageManager.remove(ofType: eventType, withStorageId: nonExistentStorageId)
+        // If no error is thrown, the test passes
+    }
+
+    // MARK: - Concurrency and Ordering Tests
+
+    func test_concurrentEventStorageAndRetrieval_givenMultipleEvents_expectAllEventsStoredAndRetrieved() async throws {
+        // Create multiple events
+        let events = (1 ... 10).map { ProfileIdentifiedEvent(identifier: "\($0)") }
+
+        // Concurrently store events
+        await withThrowingTaskGroup(of: Void.self) { group in
+            for event in events {
+                group.addTask {
+                    try await self.eventStorageManager.store(event: event)
+                }
+            }
+        }
+
+        // Load events and verify
+        let loadedEvents = try await eventStorageManager.loadEvents(ofType: ProfileIdentifiedEvent.key)
```

Comment:
on the surface that might look like useless test. However, it is right now because we are using actors. If for whatever reason that change, this test can add value.

The reality though, you can't deterministically validate thread safety. In their definition their produce inconsistent behavior :s